src/webui/api/torrentscontroller.cpp

```diff
@@ -559,12 +559,20 @@ void TorrentsController::filePrioAction()
     checkParams({"hash", "id", "priority"});
 
     const QString hash = params()["hash"];
-    int fileID = params()["id"].toInt();
-    int priority = params()["priority"].toInt();
-    BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+    const QStringList fileIDs {params()["id"].split('|')};
+    bool ok;
+    const int priority = params()["priority"].toInt(&ok);
+    if (!ok)
+        throw APIError(APIErrorType::BadParams, tr("Priority must be an int"));
 
-    if (torrent && torrent->hasMetadata())
-        torrent->setFilePriority(fileID, priority);
+    BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+    if (torrent && torrent->hasMetadata()) {
+        for (const auto &fileID : fileIDs) {
+            const int id = fileID.toInt(&ok);
+            if (ok)
```

Comment:
Why invalid file id isn't bad param error? (Unlike invalid prio)

Reply:
> Just check that they are all correct data type

We have two options: check that they're valid `int`s (which I'm already doing), or check that they're a valid `FilePriority` values.

> P.S. please, don't remove your comments even if you change your mind later. Just say about. Otherwise it's hard to maintain the discussion.

No one removed comments. You replied to the wrong comment thread...

Reply:
>No one removed comments. You replied to the wrong comment thread...

Forgive. I just got an email notification, and it was like you answered my question in this thread. But then I didn't find your answer here.